mlir/test/Dialect/Linalg/canonicalize.mlir

```diff
@@ -352,6 +352,20 @@ func.func @fold_fill_extract(%arg0 : i1) -> i1 {
 
 // -----
 
+func.func @fold_fill_extract_slice() -> tensor<2x1920x64x66xf32> {
```

Comment:
Please add a negative test, e.g. `linalg.fill` following by `tensor.insert`? Provided that doesn't trigger some other canonicalization.

Reply:
Could you elaborate why we may need such a test here? To test that such a pattern wouldn't get folded?

Reply:
>  To test that such a pattern wouldn't get folded?

Indeed. IMHO, we should always try covering "interesting" corner cases (within reason). These tests effectively document the code and its design - a negative test is an example of where the fold should not work. And if anything changes, we will almost immediately know (the test will start failing),